Formula/lilypond.rb

```diff
@@ -0,0 +1,61 @@
+class Lilypond < Formula
+  desc "Music engraving program"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  livecheck do
+    url "https://lilypond.org/source.html"
+    regex(/href=.*?lilypond[._-]v?(\d+(?:\.\d+)+)\.t/i)
+  end
+
+  depends_on "bison" => :build # Lilypond requires bison 2.4.1 or above
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "perl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "t1utils" => :build
+  depends_on "texinfo" => :build # lilypond requires texinfo 6.1 or above
+  depends_on "texlive" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "guile@2"
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+
+  def install
+    system "./configure",
+            "--prefix=#{prefix}",
+            "--datadir=#{share}",
+            "--with-texgyre-dir=#{Formula["texlive"].opt_share}/texmf-dist/fonts/opentype/public/tex-gyre",
+            "--disable-documentation"
+    ENV.prepend_path "LTDL_LIBRARY_PATH", Formula["guile@2"].lib
+    system "make"
+    system "make", "install"
+
```

Comment:
Setting `--datadir` didn't work, unfortunately.
```
lilypond:
  * Emacs Lisp files were linked directly to "/opt/homebrew/share/emacs/site-lisp".
    This may cause conflicts with other packages.
    They should instead be installed into:
      /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond
    The offending files are:
      /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-what-beat.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-words.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-song.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-font-lock.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-init.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-indent.el
            /opt/homebrew/opt/lilypond/share/emacs/site-lisp/lilypond-mode.el
```

But what we can do is this:

```suggestion

    elisp.install share.glob("emacs/site-lisp/*.el")
```
which is less fragile than the `inreplace`.

Reply:
oops, I only see your message now. Please verify if my solution I just pushed works too.

Reply:
I think we want to set `--datadir` anyway, just in case that reference is compiled into the software and used somewhere. I also slightly prefer `foo.glob` to using `Dir[]`, but that's just personal preference.

Reply:
ok, I will make these changes.